packages/wallet-adapter-react/src/useWallet.tsx

```diff
@@ -39,7 +39,7 @@ export interface WalletContextState {
   ): Promise<PendingTransactionResponse>;
   signMessage(message: SignMessagePayload): Promise<SignMessageResponse>;
   signMessageAndVerify(message: SignMessagePayload): Promise<boolean>;
-  changeNetwork(network: Network): Promise<AptosChangeNetworkOutput>;
+  changeNetwork(network: Network, chainId?: number): Promise<AptosChangeNetworkOutput>;
```

Comment:
Lets let the user pass in the chainId as a string, and have the adapter do the parsing before sending it to the wallet

Reply:
I know in the example used in this repo, the chainId is provided in an input field and using string would make sense. However, I believe this to be an exceptional case. I imagina all dApps that will utilize this feature will handle the logic in the backend, and the chainId is not provided by the user but rather from some backend storage where it's stored as number.

Let me know what you think.

Reply:
I see. I'd argue that they will be retrieving it from an API query, where it will mostly be in a string format. Instead of requiring the [dapp to convert it to a number](https://github.com/aptos-labs/aptos-wallet-adapter/pull/425/files#diff-010ca2e223c5ddff67f01665c9504e1c74c1b0c36861e9b6441e6a13f11b0675R299-R300), the [adapter can handle](https://github.com/aptos-labs/aptos-wallet-adapter/pull/425/files#diff-8a76cc616d7961765eb7ff494b4e86bc0e56cce6d0273caaa57af714ac95757eR1178) that conversion for them.

Reply:
That makes sense, thanks for clarifying. I will modify it and push the changes.